onnxruntime/core/graph/graph.cc

```diff
@@ -2029,8 +2029,10 @@ void Graph::SetDescription(const std::string& description) {
 }
 
 void Graph::AddInitializedTensor(const TensorProto& tensor) {
-  if (name_to_initial_tensor_.end() != name_to_initial_tensor_.find(tensor.name())) {
-    return;
+  auto existing = name_to_initial_tensor_.find(tensor.name());
+  if (existing != name_to_initial_tensor_.cend()) {
+    ORT_ENFORCE(existing->second == &tensor,
+                "AddInitializedTensor already has tensor with name ", tensor.name(), " but different TensorProto.");
```

Comment:
should we still exit early in this case? if not, another initializer is added with `graph_proto_->add_initializer()`.

Reply:
Yes. Fixed. Thanks.